### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by mejiaj**
## Summary

Dependency updates.

## Impacted Areas of the Site

- Local environment

## This pull request changes...
- [x] Created an NVMRC that matches supported node engine. This will help us stay on the same version of node.
- [x]  Regenerated `Gemfile.lock` file. There was a local issue with mismatched ruby deps.
- [x] Updated node minor and patch dependencies. This should resolve current dependabot alerts.

## This pull request is ready to merge when...
- [ ] Tests have been updated (and all tests are passing)
- [ ] This code has been reviewed by someone other than the original author
- [ ] The change has been documented

### CODE REVIEW

**Review by ryanwoldatwork — APPROVED** [review]
lgtm!

**Review by XavierMetichecchia — COMMENTED** [review]
Def might be speaking way out of my league here but I suspect you may have made changes to the Gemfile for the platform because you are likely developing locally on a mac. In the cloud tho it will be an x86 based OS running a Ruby container, just want to make sure this won't mess with any of that. Guess we don't really know until we try! lol

**Comment by mejiaj** [comment]
> Def might be speaking way out of my league here but I suspect you may have made changes to the Gemfile for the platform because you are likely developing locally on a mac. In the cloud tho it will be an x86 based OS running a Ruby container, just want to make sure this won't mess with any of that. Guess we don't really know until we try! lol

That makes sense. This file was updated _after_ I matched the ruby and bundler versions, so it makes sense to include anyways.

File: Gemfile.lock [hunk 1]
```diff
@@ -57,23 +58,26 @@ GEM
...
     terminal-table (3.0.2)
       unicode-display_width (>= 1.1.1, < 3)
     thor (0.20.3)
-    unicode-display_width (2.4.2)
+    unicode-display_width (2.5.0)
     webrick (1.8.1)
 
 PLATFORMS
+  arm64-darwin-23
   x86_64-darwin-21
```
Comment (mejiaj) [inline Gemfile.lock:81]: @ryanwoldatwork @XavierMetichecchia thanks for the considerations. I didn't realize the previous platform was removed via `bundle install`. 

I've reverted that change in 8e5536c and have added my local platform in 52d0489 (_via the command `bundle lock --add-platform arm64-darwin-23`_). This should improve compatibility moving forward.

**Review by XavierMetichecchia — APPROVED** [review]
Looks kosher!

**Pull request merged**